server/opts.go

```diff
@@ -2432,10 +2432,18 @@ func setBaselineOptions(opts *Options) {
 			opts.LeafNode.AuthTimeout = float64(AUTH_TIMEOUT) / float64(time.Second)
 		}
 	}
+	// Set baseline connect port for remotes.
+	for _, r := range opts.LeafNode.Remotes {
+		if r != nil && r.URL.Port() == "" {
+			r.URL.Host = fmt.Sprintf("%s:%d", r.URL.Host, DEFAULT_LEAFNODE_PORT)
```

Comment:
I don't think you should override the `r.URL.Host` this way, instead you should recreate `r.URL` with `url.Parse()`. Also use `net.JoinHostPort()` instead of Sprintf() to be able to support IPv6.

Reply:
I might agree about JoinHostPort, but we have already parsed the host correctly, we know we are missing a port. The behavior I want is a default port in that situation. So don't think I need to recreate the URL.

Reply:
Ok.

Reply:
I will change over though to Join.

Reply:
fixed